channel/channel_record_test.go

```
@@ -367,3 +381,297 @@ func TestMuxOutputLooksValid(t *testing.T) {
+		t.Fatalf("write fake ffmpeg: %v", err)
+	}
+	// Stub ffprobe to report aligned streams so this test only checks the
+	// timing-preservation flags. Offset trimming is exercised separately.
+	ffprobePath := filepath.Join(dir, "ffprobe")
+	if err := os.WriteFile(ffprobePath, []byte("#!/bin/sh\necho 0.000000\n"), 0755); err != nil {
+		t.Fatalf("write fake ffprobe: %v", err)
+	}
+	t.Setenv("PATH", dir)
+
+	oldDetectedEncoder := detectedEncoder
+	oldDetectedEncoderOnce := detectedEncoderOnce
```

golangci-lint (2.12.2)

[error] copylocks: assignment copies lock value to oldDetectedEncoderOnce: sync.Once contains sync.noCopy

(govet)